setup.py

```diff
@@ -235,27 +234,33 @@ def get_build_overrides():
     from numpy.distutils.command.build_clib import build_clib
     from numpy.distutils.command.build_ext import build_ext
 
-    def _is_using_gcc(obj):
-        is_gcc = False
+    def _is_using_old_gcc(obj):
+        is_old_gcc = False
         if obj.compiler.compiler_type == 'unix':
-            cc = sysconfig.get_config_var("CC")
-            if not cc:
-                cc = ""
-            compiler_name = os.path.basename(cc)
-            is_gcc = "gcc" in compiler_name
-        return is_gcc
+            cc = obj.compiler.compiler[0]
+            if "gcc" in cc:
+                out = subprocess.run([cc, '-dumpversion'],
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.PIPE,
+                                    )
+                # will print something like b'4.2.1\n'
+                ver_parts = out.stdout.split(b'.')
+                if int(ver_parts[0]) < 6:
```

Comment:
Or perhaps:
```suggestion
                # will print something like b'4.2.1\n'
                ver = LooseVersion(out.stdout.strip())
                if ver < "6.0.0":
```

Reply:
I think `packaging.version.parse` is to be preferred over `distutils.version.LooseVersion`, see [this comment](https://github.com/pypa/setuptools/issues/2261#issuecomment-660343237) in the issue about setuptools taking over distutils.

Reply:
changed to use LooseVersion, we can revisit if the preference becomes more widespread